Cython/Compiler/ExprNodes.py

```diff
@@ -13755,6 +13769,59 @@ def analyse_type_annotation(self, env, assigned_value=None):
             warning(annotation.pos, "Unknown type declaration in annotation, ignoring")
         return base_type, arg_type
 
+class AssignmentExpressionNode(ExprNode):
+    """
+    Also known as a named expression or the walrus operator
+
+    Arguments
+    lhs - NameNode
+    rhs - ExprNode  (not stored as an attribute because it is likely to be heavily transformed)
+
+    internally uses
+    impl - UtilNodes.TempResultFromStatNode
+    """
+    # unusually, child_attrs and subexprs are deliberately not the same because it's
+    # sort of a hybrid node
+    subexprs = ["impl"]
+
+    impl = None
+    is_temp = True
+
+    def __init__(self, pos, lhs, rhs, **kwds):
+        from .UtilNodes import TempResultFromStatNode, ResultRefNode
+        super(AssignmentExpressionNode, self).__init__(pos, lhs=lhs, **kwds)
+
+        class WalrusResultRefNode(ResultRefNode):
```

Comment:
Is there any particular reason for declaring this class inside the method? Creating classes is a fairly expensive operation, and it doesn't seem to need any closure values or otherwise require a different copy for every single node.

Reply:
It does capture `self` (which isn't great itself because it isn't hugely obvious).

It's also inherits from a local import from `UtilNodes.py` which I don't want to import directly in the main body of `ExprNodes.py` because it'd create a circular dependency.

I could always move this class to `UtilNodes` and refactor the capture if needed.